aea/cli/registry/fetch.py

```diff
@@ -44,13 +45,18 @@ def fetch_agent(ctx: Context, public_id: PublicId) -> None:
     target_folder = os.path.join(ctx.cwd, name)
     os.makedirs(target_folder, exist_ok=True)
 
+    filepath = download_file(file_url, ctx.cwd)
+    extract(filepath, target_folder)
+    ctx.cwd = target_folder
+    try_to_load_agent_config(ctx)
+
     click.echo("Fetching dependencies...")
     for item_type in ("connection", "contract", "skill", "protocol"):
         item_type_plural = item_type + "s"
         for item_public_id in resp[item_type_plural]:
             item_public_id = PublicId.from_str(item_public_id)
             try:
-                fetch_package(item_type, item_public_id, target_folder)
+                _add_item(click_context, item_type, item_public_id)
```

Comment:
if you call fetch_package, then the dependencies of a package are not fetched, with this change, they are fetched too.